zanata-war/src/main/java/org/zanata/adapter/PropertiesAdapter.java

```diff
@@ -0,0 +1,225 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+
+package org.zanata.adapter;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+import org.apache.commons.lang.StringUtils;
+import org.zanata.adapter.properties.PropReader;
+import org.zanata.adapter.properties.PropWriter;
+import org.zanata.common.ContentState;
+import org.zanata.common.LocaleId;
+import org.zanata.exception.FileFormatAdapterException;
+import org.zanata.rest.dto.resource.Resource;
+import org.zanata.rest.dto.resource.TranslationsResource;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Adapter for reading and write {@link org.zanata.common.DocumentType#PROPERTIES} file
+ *
+ * TODO: Convert to okapi properties adapter once all client conversion is
+ * migrated to server
+ *
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Slf4j
+public class PropertiesAdapter implements FileFormatAdapter {
+
+    public static final String ISO_8859_1 = Charsets.ISO_8859_1.name();
+
+    public static final String UTF_8 = Charsets.UTF_8.name();
+
+    @Override
+    public Resource parseDocumentFile(URI fileUri, LocaleId sourceLocale,
+        Optional<String> filterParams) throws FileFormatAdapterException,
+        IllegalArgumentException {
+
+        return parseDocumentFile(fileUri, sourceLocale, filterParams,
+            ISO_8859_1);
+    }
+
+    @Override
+    public TranslationsResource parseTranslationFile(URI fileUri,
+        String localeId, Optional<String> params)
+        throws FileFormatAdapterException, IllegalArgumentException {
+
+        return parseTranslationFile(fileUri, localeId, params, ISO_8859_1);
+    }
+
+    @Override
+    public void writeTranslatedFile(OutputStream output, URI originalFile,
+        Resource resource, TranslationsResource translationsResource,
+        String locale, Optional<String> params)
+        throws FileFormatAdapterException, IllegalArgumentException {
+
+        writeTranslatedFile(output, resource, translationsResource, locale,
+                params, ISO_8859_1);
+    }
+
+    public Resource parseDocumentFile(URI fileUri, LocaleId sourceLocale,
+        Optional<String> filterParams, String charset) throws FileFormatAdapterException,
+        IllegalArgumentException {
+
+        if (sourceLocale == null) {
+            throw new IllegalArgumentException("Source locale cannot be null");
+        }
+
+        if (fileUri == null) {
+            throw new IllegalArgumentException("Document URI cannot be null");
```

Comment:
- [ ] I wonder if we should add some non-null annotations to the interface for these things, so that these checks are not needed. There are some of these checks earlier in the diff that could be avoided with the right annotations too.